src/hooks/useAds.ts

```diff
@@ -0,0 +1,55 @@
+import { useQuery } from 'react-query';
+
+import { getMediaAds, getAdSchedule } from '#src/services/api.service';
+import type { AdScheduleUrls } from '#types/ad-schedule';
+
+const CACHE_TIME = 60 * 1000 * 20;
+
+const useAdSchedule = (adScheduleId: string | null | undefined, enabled: boolean) => {
+  const { isLoading, data } = useQuery(
+    ['ad-schedule', adScheduleId],
+    async () => {
+      const adSchedule = await getAdSchedule(adScheduleId);
+
+      return adSchedule;
+    },
+    { enabled, cacheTime: CACHE_TIME, staleTime: CACHE_TIME },
+  );
+
+  return {
+    isLoading,
+    data,
+  };
+};
+
+const useMediaAds = (jsonUrl: string | null | undefined, mediaId: string, enabled: boolean) => {
+  const { isLoading, data } = useQuery(
+    ['media-ads', mediaId],
+    async () => {
+      // Waiting for `prd` deploy to remove `replace`
+      const mediaAds = await getMediaAds(jsonUrl?.replace('advertising/site', 'sites') as string, mediaId);
+
+      return mediaAds;
+    },
+    { enabled, cacheTime: CACHE_TIME, staleTime: CACHE_TIME },
+  );
+
+  return {
+    isLoading,
+    data,
+  };
+};
+
+export const useAds = (adScheduleId: string | null | undefined, urls: AdScheduleUrls | undefined, mediaId: string) => {
+  const hasAdConfig = !!urls?.json;
+
+  // Fetch ad-schedule only when ad-config is not set (`adScheduleUrls.json` prop is not set) + adScheduleId is present
+  const { data: adSchedule, isLoading: isAdScheduleLoading } = useAdSchedule(adScheduleId, !!(adScheduleId && hasAdConfig));
+  // adScheduleUrls.json prop exists when ad-config is attached to the App Config
+  const { isLoading: isMediaAdsLoading, data: mediaAds } = useMediaAds(urls?.json, mediaId, hasAdConfig);
+
+  return {
```

Comment:
These ORs are kind of confusing. I would probably not worry about splitting the 2 ad lookups into separate methods and just use 1 useQuery with if/else logic inside of it.

Reply:
My idea was to provide an easy way of using just ad-schedule or ad-config if partners need id.

Reply:
+ we also need a key to cache the results. For ad-config it should be a media item, but for ad-schedule we don't need it.

Reply:
Oh I see. That's a nice consideration for our sample repo. Is there a way you can implement a boolean switch to make sure only 1 query is active at a time and that the results from the intended active query are returned based on that switch as well?  My only reservation with the current approach is worrying that somehow both queries will get triggered and return some weird combined or unexpected results

Reply:
I added a ternary operator to differentiate between two flows. In case we have a url in the config, then we consider the app to use per media ads. If it is not present - then we use old flow with an ad-schedule.

Reply:
Much better! Thanks!